src/front-door/azext_front_door/custom.py

```diff
@@ -446,42 +446,75 @@ def add_fd_backend(cmd, resource_group_name, front_door_name, backend_pool_name,
     frontdoor = client.get(resource_group_name, front_door_name)
     backend_pool = next((x for x in frontdoor.backend_pools if x.name == backend_pool_name), None)
     if not backend_pool:
-        from knack.util import CLIError
-        raise CLIError("Backend pool '{}' could not be found on frontdoor '{}'".format(
+        from azure.cli.core.azclierror import ResourceNotFoundError
+        raise ResourceNotFoundError("Backend pool '{}' could not be found on frontdoor '{}'".format(
             backend_pool_name, front_door_name))
     backend_pool.backends.append(backend)
     client.begin_create_or_update(resource_group_name, front_door_name, frontdoor).result()
     return backend
 
 
+def update_fd_backend(cmd, resource_group_name, front_door_name, backend_pool_name, index, address=None,
+                      http_port=None, https_port=None, disabled=None, priority=None, weight=None,
+                      backend_host_header=None, private_link_alias=None, private_link_resource_id=None,
+                      private_link_location=None, private_link_approval_message=None):
+
+    client = cf_frontdoor(cmd.cli_ctx, None)
+    frontdoor = client.get(resource_group_name, front_door_name)
+    backend_pool = next((x for x in frontdoor.backend_pools if x.name == backend_pool_name), None)
+    if not backend_pool:
+        from azure.cli.core.azclierror import ResourceNotFoundError
+        raise ResourceNotFoundError("Backend pool '{}' could not be found on frontdoor '{}'".format(
+            backend_pool_name, front_door_name))
+    if index > len(backend_pool.backends) or index <= 0:
+        from azure.cli.core.azclierror import InvalidArgumentValueError
+        raise InvalidArgumentValueError("Backend range is from 1 to {}, index '{}' could not be found on frontdoor '{}'".format(
+            len(backend_pool.backends), index, front_door_name))
+    backend = backend_pool.backends[index - 1]
+    with UpdateContext(backend) as c:
+        c.update_param('address', address, None)
+        c.update_param('http_port', http_port, None)
+        c.update_param('https_port', https_port, None)
+        c.update_param('enabled_state', 'Disabled' if disabled else 'Enabled', None)
+        c.update_param('priority', priority, None)
+        c.update_param('weight', weight, None)
+        c.update_param('backend_host_header', backend_host_header, None)
+        c.update_param('private_link_alias', private_link_alias, None)
+        c.update_param('private_link_resource_id', private_link_resource_id, None)
+        c.update_param('private_link_location', private_link_location, None)
+        c.update_param('private_link_approval_message', private_link_approval_message, None)
+    client.begin_create_or_update(resource_group_name, front_door_name, frontdoor).result()
+    return backend
```

Comment:
Why the return value is not from request result?

Reply:
Based on `add_fd_backend`, backend is a part of frontdoor, we just need return backend, not all result from frontdoor.

Reply:
I think we should extract the backend from the result again instead of returning the backend object used in request construction. Because the backend in result may be not the same as the backend in request. For example, some read only property may be changed in result.